### PR TITLE
Issue #058 / #062 - Add Private to QueueEntryEntity / Submit a Question to the Bot

### LINKED ISSUE #58: Submit a question to the bot
When a user joins the queue, give them the option to also submit a question along when they enter.

### LINKED ISSUE #62: Add Private to QueueEntryEntity
add a boolean to mark a queue entry entity as private

### LINKED ISSUE #67: Undefined userID and resolved status in queue members
It appears that `userId` and `resolved` status for userID's are not making their way onto the in-app Queue Datastruture. After investigation, it appears that the values are indeed being inserted into the DB correctly, however, the code currently generates a QueueEntry using the results returned from the [SQL Insert operation](https://github.com/beaver-hours/BeaverHours/blob/82f84518caa8b37005c82a049fdef009933a9d1c/bot/api/addQueueEntryToDb.ts#L23), which may not return all fields and thus from within the app, all members of the queue appear to be unidentifiable. 

NOTE: This may appear to happen with other fields as well, such as `question`, however it was not checked at the time of this bug being found.

Issue Appearing within Teams:
image

Values Persisted correctly in DB:
image

SQL Insert return missing `userId` and `resolved` values:
image

### PR DESCRIPTION
**Pull request by richardcoucoules**
Closes #58 
Closes #62 

This PR adds the `privateEntry` field to the `QueueEntry` and `QueueEntryEntity` classes. Students can join a queue privately by using the `private join office hours` command. In addition, students can supply a question by appending it to the end of the `join office hours` or `private join office hours` command. Private entries have their question masked in all places where the queue is displayed to users, but an optional parameter to the `toString` method of the `QueueEntry` class and `entriesToString` method of the `Queue` class were added to allow the private questions to be unmasked.

image

### CODE REVIEW

**Review by Jaqito — CHANGES_REQUESTED** [review]
Awesome work! Only thing I'll suggest is that the chat prompt be updated to reflect the new options for messaging. A user wouldn't know right now that they can enter the chat privately or that they can submit a message

File: tabs/package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "@types/react": "^16.14.6",
     "@types/react-dom": "^16.9.12",
     "@types/react-router-dom": "^5.1.7",
+    "chokidar": "^3.5.3",
```
Comment (Jaqito) [inline tabs/package.json:23]: what's this for?

Comment (richardcoucoules) [reply tabs/package.json:23]: Without this, I was getting compilation issues on my Windows machine. I searched my issue and it seems it has something to do with npm and potentially may be Windows-only. Chokidar is a React dependency and npm was having trouble auto-resolving it due to versioning conflicts somewhere. Adding this line effectively made the dependency explicit and stopped the errors on my end.

**Review by Jaqito — APPROVED** [review]
Nice work!

**Review by rorochaudhary — APPROVED** [review]
Great job! The bug fix for certain `QueueEntry` fields not making it in the queue was a good catch as it is important for the overall functionality of the bot.

**Pull request merged**